### PR TITLE
Task #211309 Added Program_id and academic_year_id in facilitator and edit_request…

### PR DESCRIPTION
**Pull request by Tusharmahajan12**
… module
https://tracker.tekdi.net/issues/211309
# I have ensured that following `Pull Request Checklist` is taken care of before sending this PR

* [ ] Code is formatted as per format decided
* [ ] Updated acceptance criteria in tracker
* [ ] Updated test cases in test-cases-tracker
* [ ] Updated new API endpoint if any in common postman collection
* [ ] Updated DB changes in db-tracker if any

## Summary by CodeRabbit

- **New Features**
	- Enhanced the facilitator module to support more specific queries by including program and academic year identifiers.
	- Streamlined edit request retrieval processes for both general and administrative users, improving efficiency and relevance.

- **Refactor**
	- Optimized middleware exclusions for better route management within the facilitator module.
	- Simplified parameter lists in edit request service methods for improved usability.

### CODE REVIEW

File: src/src/facilitator/facilitator.module.ts [hunk 1]
```diff
@@ -42,47 +42,31 @@ export class FacilitatorModule implements NestModule {
...
 				},
 				{ path: '/facilitators/:id', method: RequestMethod.PATCH },
-				'/facilitators/admin/okyc_details_override',
 				'/facilitators/admin/search-by-ids',
-				'/facilitators/exportCsv',
 				'/facilitators/program-facilitator/add',
-				'/facilitators/admin/filter-by-beneficiaries',
 				{
 					path: '/facilitators/update-facilitator-aadhar/:id',
 					method: RequestMethod.PATCH,
```
[the remark this answers is not in the document]
Comment (GitVivekHub) [reply src/src/facilitator/facilitator.module.ts:68]: check this

File: src/src/facilitator/facilitator.service.ts [hunk 2]
```diff
@@ -1866,6 +1875,7 @@ export class FacilitatorService {
 	  }`,
 			variables: variables,
 		};
+		console.log('ss', data.query);
```
Comment (GitVivekHub) [inline src/src/facilitator/facilitator.service.ts:1878]: remove console.log

Comment (Tusharmahajan12) [reply src/src/facilitator/facilitator.service.ts]: removed

**Review by GitVivekHub — APPROVED** [review]

File: src/src/facilitator/facilitator.service.ts [hunk 3]
```diff
@@ -1223,7 +1231,7 @@ export class FacilitatorService {
 			}
 
 			filterQueryArray.unshift(
-				`{program_faciltators: {id: {_is_null: false}, parent_ip: {_eq: "${user?.data?.program_users[0]?.organisation_id}"}}}`,
+				`{program_faciltators: {id: {_is_null: false}, parent_ip: {_eq: "${user?.data?.program_users[0]?.organisation_id}"},academic_year_id:{_eq:${academic_year_id}},,program_id:{_eq:${program_id}}}}`,
```
Comment (manojLondhe) [inline src/src/facilitator/facilitator.service.ts:1234]: one comma extra? `,,`

Comment (Tusharmahajan12) [reply src/src/facilitator/facilitator.service.ts]: sorry removed

**Pull request merged**